session8/src/com/itschool/homework/Session8Challenge9.java

```diff
@@ -8,18 +10,20 @@ public static void main(String[] args) {
 
     public void challengeRun9() {
         StringBuilder text = new StringBuilder("Let's gooo!");
+        HashSet<Character> processedChars = new HashSet<>();
```

Comment:
nice! Can you say how the `HashSet` works?

Reply:
HashSet stores unique elements. By it's name, each element is stored with a unique hash code. In case more elements have the same hash code, it stores them in a bucket before checking next element and store it. This way it handles collisions.

Reply:
Partially true. To mention that in all situations the elements are stored in buckets, not just the ones with collision.